src/RefractDataStructure.cc

```diff
@@ -618,10 +646,10 @@ namespace drafter {
 
                 refract::ArrayElement* a = new refract::ArrayElement;
 
+                ElementInfoToElement<T> fetch;
+
                 for (auto sample : samples) {
```

Comment:
Please refactor this to use either `auto&` or `const auto&`. It isn't good practice to use mere `auto` in ranged for loops because of implicit copying.